manager/src/main/java/org/openremote/manager/mqtt/AssetInterceptHandler.java

```diff
@@ -0,0 +1,190 @@
+import java.util.Map;
+import java.util.logging.Logger;
+
+import static org.openremote.manager.mqtt.MqttBrokerService.TOPIC_SEPARATOR;
+import static org.openremote.model.Constants.KEYCLOAK_CLIENT_ID;
+
+public class AssetInterceptHandler implements InterceptHandler {
+
+    private static final Logger LOG = Logger.getLogger(AssetInterceptHandler.class.getName());
+
+    protected final ManagerIdentityService identityService;
+    protected final ManagerKeycloakIdentityProvider identityProvider;
+    protected final AssetStorageService assetStorageService;
+    protected final AssetProcessingService assetProcessingService;
+    protected final MessageBrokerService messageBrokerService;
+    protected final MqttConnector mqttConnector;
+    protected final BiConsumer<String, AttributeEvent> attributeEventConsumer;
+
+    AssetInterceptHandler(AssetStorageService assetStorageService,
+                          AssetProcessingService assetProcessingService,
+                          ManagerIdentityService managerIdentityService,
+                          ManagerKeycloakIdentityProvider managerKeycloakIdentityProvider,
+                          MessageBrokerService messageBrokerService, MqttConnector mqttConnector, BiConsumer<String, AttributeEvent> attributeEventConsumer) {
+
+        this.assetStorageService = assetStorageService;
+        this.assetProcessingService = assetProcessingService;
+        this.identityService = managerIdentityService;
+        this.identityProvider = managerKeycloakIdentityProvider;
+        this.messageBrokerService = messageBrokerService;
+        this.mqttConnector = mqttConnector;
+        this.attributeEventConsumer = attributeEventConsumer;
+    }
+
+    @Override
+    public String getID() {
+        return UniqueIdentifierGenerator.generateId(AssetInterceptHandler.class.getName());
+    }
+
+    @Override
+    public Class<?>[] getInterceptedMessageTypes() {
+        return InterceptHandler.ALL_MESSAGE_TYPES;
+    }
+
+    @Override
+    public void onConnect(InterceptConnectMessage interceptConnectMessage) {
+        MqttConnector.MqttConnection connection = mqttConnector.createConnection(interceptConnectMessage.getClientID(), interceptConnectMessage.getUsername(), interceptConnectMessage.getPassword());
+        String suppliedClientSecret = new String(interceptConnectMessage.getPassword(), StandardCharsets.UTF_8);
+        connection.accessToken = identityProvider.getKeycloak().getAccessToken(connection.realm, new ClientCredentialsAuthFrom(connection.username, suppliedClientSecret)).getToken();
+    }
+
+    @Override
+    public void onDisconnect(InterceptDisconnectMessage interceptDisconnectMessage) {
+        MqttConnector.MqttConnection connection = mqttConnector.removeConnection(interceptDisconnectMessage.getClientID());
+        if (connection != null) {
+            CancelEventSubscription<AttributeEvent> cancelEventSubscription = new CancelEventSubscription<>(AttributeEvent.class);
```

Comment:
Should use `EventSubscriptions:cancelAll()` to remove any subscriptions this client has made